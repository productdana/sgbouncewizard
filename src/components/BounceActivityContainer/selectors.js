import {
  selectSelectorGenerator,
  writeSelectorGenerator,
} from "../selectorHelpers";

// These test hooks will eventually turn into the following:
// - Selectors (CSS Selector strings) i.e. "[data-test=emailInput]"
// - Write Selectors (Selector objects spread onto React components) i.e. { data-test: "emailInput" }
// What if we had to update a data-test attribute value or other CSS selectors?
// - We would have to update all the test file strings in Jest/Enzyme and Cypress tests and React components.
// - This pattern helps us maintain this in the long run and update as few places as possible for CSS selector changes
const testHooks = {
  page: "activity-log-page",
  breadcrumb: "activity-log-breadcrumb",
  csvButton: "export-csv-button",
  activityFilter: "rule-filter",
  activityTable: "activity-table",
  emptyRulesWarning: "empty-rules-warning",
  pagination: "pagination",
  detailsModal: "actvity-details-modal",
};

// Selectors like "[data-test=emailInput]" to be imported into the following files:
// - Jest/Enzyme tests to be placed into .find("[data-test=emailInput]") and other similar functions
// - Cypress tests to be placed into cy.get("[data-test=emailInput]"), cy.contains(), etc.
const Selectors = selectSelectorGenerator(testHooks);

// Write Selectors like { emailInput: { data-test: "emailInput" }, ... } to be imported into
// React component files to spread the data-test attributes onto elements we would like to keep track of for tests
// i.e. <Component {...WriteSelectors.emailInput} /> => <Component data-test="emailInput" />
const WriteSelectors = writeSelectorGenerator(testHooks);

export { Selectors, WriteSelectors };
