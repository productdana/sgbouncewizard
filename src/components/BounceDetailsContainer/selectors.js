import {
  selectSelectorGenerator,
  writeSelectorGenerator,
} from "../selectorHelpers";

// These test hooks will eventually turn into the following:
// - Selectors (CSS Selector strings) i.e. "[data-test=emailInput]"
// - Write Selectors (Selector objects spread onto React components) i.e. { data-test: "emailInput" }
// What if we had to update a data-test attribute value or other CSS selectors?
// - We would have to update all the test file strings in Jest/Enzyme and Cypress tests and React components.
// - This pattern helps us maintain this in the long run and update as few places as possible for CSS selector changes
const testHooks = {
  details: "details",
  detailsEditable: "details-editable",
  changelog: "changelog",
  editButton: "edit-button",
  saveButton: "save-button",
  cancelButton: "cancel-button",
  description: "description",
  bounceId: "bounce_id",
  responseCode: "response-code",
  enhancedCode: "enhanced-code",
  regex: "regex",
  priority: "priority",
  bounceAction: "bounce-action",
  changelogModal: "changelog-modal",
  confirmModal: "commit-confirm-modal",
  saveConfirmationModal: "save-confirmation-modal",
  cancelConfirmationModal: "cancel-confirmation-modal",
  confirmationSubmit: "create-confirm-submit",
  commitInput: "commit-input",
  confirmSubmit: "confirm-submit",
  pagination: "pagination",
};

// Selectors like "[data-test=emailInput]" to be imported into the following files:
// - Jest/Enzyme tests to be placed into .find("[data-test=emailInput]") and other similar functions
// - Cypress tests to be placed into cy.get("[data-test=emailInput]"), cy.contains(), etc.
const Selectors = selectSelectorGenerator(testHooks);

// Write Selectors like { emailInput: { data-test: "emailInput" }, ... } to be imported into
// React component files to spread the data-test attributes onto elements we would like to keep track of for tests
// i.e. <Component {...WriteSelectors.emailInput} /> => <Component data-test="emailInput" />
const WriteSelectors = writeSelectorGenerator(testHooks);

export { Selectors, WriteSelectors };
