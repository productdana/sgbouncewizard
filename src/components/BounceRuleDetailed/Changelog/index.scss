table {
  table-layout: fixed;

  .row-date {
    width: 25%;
  }

  .row-user {
    width: 15%;
  }

  .row-commit {
    width: 40%;
  }

  .row-actions {
    width: 20%;
  }
}
