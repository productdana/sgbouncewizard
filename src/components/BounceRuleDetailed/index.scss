.detail-container {
  display: inline-block;
  justify-content: center;
  align-items: center;
  padding: 5px;
  background: #f4f6f7;

  td {
    padding: 15px 10px;

    &.description-cell {
      width: 25%;
    }
  }

  th,
  td {
    word-wrap: break-word;
  }

  &:hover {
    cursor: pointer;
    box-shadow: 0.5px 0 5px #9e9e9e;
    .editable {
      display: block;
    }
  }

  .table-wrap {
    margin-bottom: 0px;
  }

  .editable {
    position: absolute;
    padding-top: 12px;
    padding-right: 12px;
    top: 0;
    right: 0;
    display: none;

    .sg-icon {
      color: #294661;
      font-size: 30px;
    }
  }

  &.detail-container-editable {
    box-shadow: 0.5px 0 5px #9e9e9e;

    .input-text-wrap {
      margin-bottom: 0;
    }
  }
}

.sg-button {
  width: 125px;
}

.sm-sg-button {
  width: 100px;
  margin: 10px;
}

.details-button-column {
  margin-top: 10px;
}

.table-fixed {
  table-layout: fixed;
}

.description-info {
  width: 100%;
  .table-wrap {
    margin-bottom: 0;
  }
}

.detail-info {
  display: flex;
  width: 100%;
}

.changelog-container {
  margin-top: 2vh;

  th.actions-align {
    padding-right: 0;
    text-align: right;
  }

  .changelog-csv {
    font-size: 15px;
    display: flex;
    align-items: center;
    margin: 0;
    grid-column: 12 / span 1;
  }

  h2 {
    margin-bottom: 5px;
  }

  .currentBadge {
    position: absolute;
    left: -55px;
    margin-top: 3px;
    background-color: #489be8;
    -webkit-transform: rotate(-90deg);
    -moz-transform: rotate(-90deg);
    -o-transform: rotate(-90deg);
    -ms-transform: rotate(-90deg);
    transform: rotate(-90deg);
  }
}

.changelog-view-icon-cell {
  text-align: right;
}

.changelog-view-icon {
  color: #7f90a0;
  &:hover {
    cursor: pointer;
  }
}

.changelog-modal {
  padding: 2vh;

  .table-wrap td,
  .table-wrap th {
    padding: 0.8rem;
  }

  h1 {
    margin-bottom: 0px;
  }
  .change-table {
    > td {
      padding: 10rem;
    }
  }

  .changelog-modal-button {
    text-align: right;
  }
}

.uneditable {
  padding: 10px;
}

.inuse {
  background: #f4f6f7;
  color: #489be8;
}
