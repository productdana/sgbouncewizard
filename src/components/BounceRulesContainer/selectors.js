import {
  selectSelectorGenerator,
  writeSelectorGenerator,
} from "../selectorHelpers";

// These test hooks will eventually turn into the following:
// - Selectors (CSS Selector strings) i.e. "[data-test=emailInput]"
// - Write Selectors (Selector objects spread onto React components) i.e. { data-test: "emailInput" }
// What if we had to update a data-test attribute value or other CSS selectors?
// - We would have to update all the test file strings in Jest/Enzyme and Cypress tests and React components.
// - This pattern helps us maintain this in the long run and update as few places as possible for CSS selector changes
const testHooks = {
  page: "bounce-rules-page",
  breadcrumb: "bounce-rules-breadcrumb",
  csvButton: "export-csv-button",
  createRuleButton: "create-rule-button",
  cancelCreateRuleButton: "cancel-create-rule-button",
  bounceRule: "bounce-rule",
  ruleFilter: "rule-filter",
  ruleTable: "rule-table",
  emptyRulesWarning: "empty-rules-warning",
  deleteConfirmation: "delete-confirmation",
  deleteConfirmationConfirm: "delete-confirmation-confirm",
  deleteAlert: "delete-alert",
  deleteButton: "delete-button",
  createRuleModal: "create-rule-modal",
  priority: "create-priority",
  description: "create-description",
  bounceAction: "create-bounce-action",
  responseCode: "create-response-code",
  enhancedCode: "create-enhanced-code",
  regex: "create-regex",
  submitButton: "create-submit-button",
  confirmModal: "commit-confirm-modal",
  confirmationSubmit: "create-confirm-submit",
  commitInput: "commit-message",
  cancelConfirmationSubmit: "cancel-confirm-submit",
  emptyCommitAlert: "empty-commit-alert",
  pagination: "pagination",
  confirmSubmit: "confirm-submit",
  invalidInput: "invalid-input",
};

// Selectors like "[data-test=emailInput]" to be imported into the following files:
// - Jest/Enzyme tests to be placed into .find("[data-test=emailInput]") and other similar functions
// - Cypress tests to be placed into cy.get("[data-test=emailInput]"), cy.contains(), etc.
const Selectors = selectSelectorGenerator(testHooks);

// Write Selectors like { emailInput: { data-test: "emailInput" }, ... } to be imported into
// React component files to spread the data-test attributes onto elements we would like to keep track of for tests
// i.e. <Component {...WriteSelectors.emailInput} /> => <Component data-test="emailInput" />
const WriteSelectors = writeSelectorGenerator(testHooks);

export { Selectors, WriteSelectors };
