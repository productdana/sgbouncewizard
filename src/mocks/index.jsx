export const mockChangelog = [
  [
    {
      id: 503,
      response_code: 450,
      enhanced_code: "4.7.0",
      regex: "Youremailingisbeingverified",
      priority: 1,
      description: "finishdeferralsecmailfiltering",
      bounce_action: "retry",
      user_id: 1,
      created_at: 280,
      comment: "Inital Setup",
      operation: "Create",
    },
  ],
];

export const mockActivityLog = [
  {
    id: 173,
    response_code: 421,
    enhanced_code: "sdfsfdfadf",
    regex: "fewgfgsdfg",
    priority: 1,
    description: "RFC5321Servicenotavailable",
    bounce_action: "retry",
    user_id: 1,
    created_at: 1,
    comment: "Inital Setup",
    operation: "Create",
  },
  {
    id: 174,
    response_code: 450,
    enhanced_code: "sadfasdf",
    regex: "asdfsdfd",
    priority: 0,
    description: "RFC5321Mailboxunavailable",
    bounce_action: "retry",
    user_id: 1,
    created_at: 2,
    comment: "Inital Setup",
    operation: "Create",
  },
  {
    id: 175,
    response_code: 451,
    enhanced_code: "sdfsdf",
    regex: "sdfasf",
    priority: 1,
    description: "RFC5321Localerrorinprocessing",
    bounce_action: "retry",
    user_id: 1,
    created_at: 3,
    comment: "Inital Setup",
    operation: "Create",
  },
  {
    id: 176,
    response_code: 452,
    enhanced_code: "sdfgerg",
    regex: "dsfgsdfg",
    priority: 1,
    description: "RFC5321Insufficientsystemstorage",
    bounce_action: "retry",
    user_id: 1,
    created_at: 4,
    comment: "Inital Setup",
    operation: "Create",
  },
  {
    id: 177,
    response_code: 454,
    enhanced_code: "sdfgsdfg",
    regex: "dfgdfg",
    priority: 1,
    description: "RFC3207TLSnotavailable",
    bounce_action: "retry",
    user_id: 1,
    created_at: 5,
    comment: "Inital Setup",
    operation: "Create",
  },
  {
    id: 178,
    response_code: 455,
    enhanced_code: "sdfgreg",
    regex: "sdfgsdfg",
    priority: 0,
    description: "RFC5321Serverunabletoaccomodateparameters",
    bounce_action: "retry",
    user_id: 1,
    created_at: 6,
    comment: "Inital Setup",
    operation: "Create",
  },
  {
    id: 179,
    response_code: 500,
    enhanced_code: "dfgsdfgre",
    regex: "dsfgsd",
    priority: 0,
    description: "RFC5321Syntaxerrorcommandunrecognized",
    bounce_action: "no_action",
    user_id: 1,
    created_at: 7,
    comment: "Inital Setup",
    operation: "Create",
  },
  {
    id: 180,
    response_code: 501,
    enhanced_code: "sdfgdsfg",
    regex: "dfgsfg",
    priority: 1,
    description: "RFC5321Syntaxerrorinparametersorarguments",
    bounce_action: "no_action",
    user_id: 1,
    created_at: 8,
    comment: "Inital Setup",
    operation: "Create",
  },
  {
    id: 181,
    response_code: 502,
    enhanced_code: "fdgert",
    regex: "sdfg",
    priority: 1,
    description: "RFC5321Commandnotimplemented",
    bounce_action: "no_action",
    user_id: 1,
    created_at: 9,
    comment: "Inital Setup",
    operation: "Create",
  },
];

export const mockBounceRules = [
  {
    id: 504,
    response_code: 551,
    enhanced_code: "sdfg",
    regex: "no MX record for domain",
    priority: 1,
    description: "new rule",
    bounce_action: "no_action",
    comment: "Inital Setup",
    created_at: 1,
  },
  {
    id: 173,
    response_code: 421,
    enhanced_code: "sdfg",
    regex: "dsfg",
    priority: 1,
    description: "RFC5321Servicenotavailable",
    bounce_action: "retry",
    user_id: 1,
    created_at: 1,
    comment: "Inital Setup",
  },
  {
    id: 505,
    response_code: 550,
    enhanced_code: "erwt",
    regex: "no MX record for domain",
    priority: 1,
    description:
      "bWFpbmx5IGxpYmVydHkgZG9tYWluIGJsb2NrIHNlZWluZyB+NTAlIG9mIGFkZHJlc3NlcyBlbmdhZ2luZyBTRyB3aWRl",
    bounce_action: "no_action",
    created_at: 1,
    comment: "Inital Setup",
  },
];

export default mockActivityLog;
