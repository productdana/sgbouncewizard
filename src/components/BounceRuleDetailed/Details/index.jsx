import React from "react";
import { TextInput } from "@sendgrid/ui-components/text-input";
import {
  Table,
  TableBody,
  TableCell,
  TableRow,
} from "@sendgrid/ui-components/table/table";
import { WriteSelectors } from "../selectors";
import { shouldDisplay } from "../../../utils/utils";

const DetailsContainer = ({ currentRule, handleEditClicked }) => {
  const {
    description,
    id,
    response_code: responseCode,
    enhanced_code: enhancedCode,
    bounce_action: bounceAction,
    regex,
    priority,
  } = currentRule;
  return (
    <div
      {...WriteSelectors.details}
      onClick={handleEditClicked}
      id="isEditClicked"
      onKeyDown={handleEditClicked}
      role="searchbox"
      tabIndex={0}
      className="detail-container card "
    >
      <div className="editable">
        <i className="sg-icon sg-icon-editor-design" />
      </div>
      <div className="description-info">
        <Table className="table-fixed">
          <TableBody>
            <TableRow>
              <TableCell className="description-cell">
                <strong>Description</strong>
              </TableCell>
              <TableCell> {shouldDisplay(description)} </TableCell>
            </TableRow>
          </TableBody>
        </Table>
      </div>
      <div className="detail-info">
        <Table>
          <TableBody>
            <TableRow>
              <TableCell>
                <strong>Bounce ID</strong>
              </TableCell>
              <TableCell> {shouldDisplay(id)} </TableCell>
            </TableRow>
            <TableRow>
              <TableCell>
                <strong>Response Code</strong>
              </TableCell>
              <TableCell> {shouldDisplay(responseCode)} </TableCell>
            </TableRow>
            <TableRow>
              <TableCell>
                <strong>Enhanced Code</strong>
              </TableCell>
              <TableCell> {shouldDisplay(enhancedCode)} </TableCell>
            </TableRow>
          </TableBody>
        </Table>
        <Table>
          <TableBody>
            <TableRow>
              <TableCell>
                <strong>Regex</strong>
              </TableCell>
              <TableCell> {shouldDisplay(regex)} </TableCell>
            </TableRow>
            <TableRow>
              <TableCell>
                <strong>Priority</strong>
              </TableCell>
              <TableCell> {shouldDisplay(priority)} </TableCell>
            </TableRow>
            <TableRow>
              <TableCell>
                <strong>Bounce Action</strong>
              </TableCell>
              <TableCell> {shouldDisplay(bounceAction)} </TableCell>
            </TableRow>
          </TableBody>
        </Table>
      </div>
    </div>
  );
};

export const DetailsContainerEditable = ({
  onChangeRule,
  onChangeRuleInt,
  updatedRule,
}) => {
  const {
    description,
    id,
    response_code: responseCode,
    enhanced_code: enhancedCode,
    bounce_action: bounceAction,
    regex,
    priority,
  } = updatedRule;
  return (
    <div
      className="detail-container detail-container-editable card "
      {...WriteSelectors.detailsEditable}
    >
      <div className="editable">
        <i className="sg-icon sg-icon-editor-design" />
      </div>
      <div className="description-info">
        <Table className="table-fixed">
          <TableBody>
            <TableRow>
              <TableCell className="description-cell">
                <strong>Description</strong>
              </TableCell>
              <TableCell>
                <TextInput
                  {...WriteSelectors.description}
                  onChange={onChangeRule}
                  id="description"
                  value={description}
                  type="text"
                />
              </TableCell>
            </TableRow>
          </TableBody>
        </Table>
      </div>
      <div className="detail-info">
        <Table>
          <TableBody>
            <TableRow>
              <TableCell>
                <strong>Bounce ID</strong>
              </TableCell>
              <TableCell>
                <div className="uneditable">
                  <strong>{id}</strong>
                </div>
              </TableCell>
            </TableRow>
            <TableRow>
              <TableCell>
                <strong>Response Code</strong>
              </TableCell>
              <TableCell>
                <TextInput
                  {...WriteSelectors.responseCode}
                  onChange={onChangeRuleInt}
                  id="response_code"
                  value={responseCode}
                  type="number"
                />
              </TableCell>
            </TableRow>
            <TableRow>
              <TableCell>
                <strong>Enhanced Code</strong>
              </TableCell>
              <TableCell>
                <TextInput
                  {...WriteSelectors.enhancedCode}
                  onChange={onChangeRule}
                  id="enhanced_code"
                  value={enhancedCode}
                  type="text"
                />
              </TableCell>
            </TableRow>
          </TableBody>
        </Table>
        <Table>
          <TableBody>
            <TableRow>
              <TableCell>
                <strong>Regex</strong>
              </TableCell>
              <TableCell>
                <TextInput
                  {...WriteSelectors.regex}
                  onChange={onChangeRule}
                  id="regex"
                  value={regex}
                  type="text"
                />
              </TableCell>
            </TableRow>
            <TableRow>
              <TableCell>
                <strong>Priority</strong>
              </TableCell>
              <TableCell>
                <TextInput
                  {...WriteSelectors.priority}
                  onChange={onChangeRuleInt}
                  id="priority"
                  value={priority}
                  type="number"
                />
              </TableCell>
            </TableRow>
            <TableRow>
              <TableCell>
                <strong>Bounce Action</strong>
              </TableCell>
              <TableCell>
                <TextInput
                  {...WriteSelectors.bounceAction}
                  onChange={onChangeRule}
                  id="bounce_action"
                  value={bounceAction}
                  type="text"
                />
              </TableCell>
            </TableRow>
          </TableBody>
        </Table>
      </div>
    </div>
  );
};

export default DetailsContainer;
