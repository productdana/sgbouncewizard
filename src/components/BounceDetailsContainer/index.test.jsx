import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import BounceRuleDetailed from ".";
import { Selectors } from "./selectors";
import { mockBounceRules, mockChangelog } from "../../mocks/index";

const {
  details,
  detailsEditable,
  changelog,
  editButton,
  cancelButton,
  saveButton,
  cancelConfirmationModal,
  changelogModal,
  pagination,
  confirmModal,
} = Selectors;

const [sampleRule] = mockBounceRules;
const [sampleChangelog] = mockChangelog;

describe("Bounce Rule Detailed", () => {
  let props;
  let mountedDetailsPage;
  const DetailsPage = () => {
    if (!mountedDetailsPage) {
      mountedDetailsPage = shallow(<BounceRuleDetailed {...props} />);
    }
    return mountedDetailsPage;
  };
  beforeEach(() => {
    props = {
      currentRule: sampleRule,
      changelog: [sampleChangelog],
      filteredChangelog: [sampleChangelog],
    };
  });

  it("should render correctly", () => {
    expect(
      renderer.create(
        <Router>
          <BounceRuleDetailed {...props} />
        </Router>
      )
    ).toMatchSnapshot();
  });

  describe("When a user visits the detailed rule page", () => {
    it("should render default state", () => {
      expect(DetailsPage().find(details)).toHaveLength(1);
      expect(DetailsPage().find(changelog)).toHaveLength(1);
      expect(DetailsPage().find(editButton)).toHaveLength(1);
      expect(DetailsPage().find(pagination)).toHaveLength(1);
    });
  });

  describe("When there are no rules", () => {
    beforeEach(() => {
      DetailsPage().setProps({ changelog: [] });
    });

    it("should not render pagination when no rules", () => {
      expect(DetailsPage().find(pagination)).toHaveLength(0);
    });
  });

  describe("When a user is editing a rule", () => {
    it("should show cancel/save button when edit is clicked", () => {
      DetailsPage().setProps({ isEditClicked: true });
      expect(DetailsPage().find(cancelButton)).toHaveLength(1);
      expect(DetailsPage().find(saveButton)).toHaveLength(1);
    });

    it("should be editable when edit is clicked", () => {
      DetailsPage().setProps({ isEditClicked: true });
      expect(DetailsPage().find(detailsEditable)).toHaveLength(1);
    });

    it("should display cancel confirm modal when cancel is clicked", () => {
      DetailsPage().setProps({ isCancelConfirmOpen: true });
      expect(DetailsPage().find(cancelConfirmationModal)).toHaveLength(1);
    });

    it("should display confirmation modal when save is clicked", () => {
      DetailsPage().setProps({ isConfirmOpen: true });
      expect(DetailsPage().find(confirmModal)).toHaveLength(1);
    });
  });

  describe("When a user views a change from the changelog", () => {
    it("should display change modal when rule is clicked", () => {
      DetailsPage().setProps({ isChangeModalOpen: true });
      expect(DetailsPage().find(changelogModal)).toHaveLength(1);
    });
  });
});
