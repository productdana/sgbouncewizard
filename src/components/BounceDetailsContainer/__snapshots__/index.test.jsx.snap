// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bounce Rule Detailed should render correctly 1`] = `
<div>
  <div
    className="header-container"
  >
    <div
      className="mako-grid__row"
    >
      <div
        className="mako-grid__col-3 col-start-2"
      >
        <img
          alt="sendgrid-logo"
          className="sg-logo"
          data-test="sg-logo"
          src="https://uiux.s3.amazonaws.com/toggleable-logos/header-logo.svg"
          style={
            Object {
              "maxWidth": "150px",
            }
          }
        />
      </div>
      <div
        className="mako-grid__col-2 col-start-10"
      >
        <div
          className="header-user-container"
        >
          <span
            className="header-greeting"
            data-test="user-greeting"
          >
            Hello, 
            User
            !
          </span>
          <a
            className="header-logout"
            data-test="logout-button"
            role="button"
            tabIndex={0}
          >
            Logout
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="mako-grid__row"
  >
    <div
      className="mako-grid__col-6 col-start-2"
    >
      <ol
        className="breadcrumb"
      >
        <li>
          <a
            href="/bounce_rules"
            onClick={[Function]}
          >
            Bounce Rules
          </a>
        </li>
        <li>
          <a
            href="/bounce_rules/504"
            onClick={[Function]}
          >
            504
          </a>
        </li>
        <li />
      </ol>
    </div>
  </div>
  <div
    className="mako-grid__row"
  >
    <div
      className="mako-grid__col-6 col-start-2"
    >
      <h1>
        Bounce Rule 
        504
      </h1>
    </div>
    <div
      className="mako-grid__col-1 col-start-11 details-button-column"
    >
      <span>
        <button
          className="btn btn-primary sg-button edit-button"
          data-test="edit-button"
          disabled={false}
          id="isEditClicked"
          onClick={[MockFunction]}
          onKeyDown={[MockFunction]}
          type="button"
        >
          Edit Rule
        </button>
      </span>
    </div>
  </div>
  <div
    className="mako-grid__row"
  >
    <div
      className="mako-grid__col-10 col-start-2"
    >
      <div>
        <div
          className="detail-container card "
          data-test="details"
          id="isEditClicked"
          onClick={[MockFunction]}
          onKeyDown={[MockFunction]}
          role="searchbox"
          tabIndex={0}
        >
          <div
            className="editable"
          >
            <i
              className="sg-icon sg-icon-editor-design"
            />
          </div>
          <div
            className="description-info"
          >
            <table
              className="table-wrap table-fixed"
            >
              <tbody>
                <tr>
                  <td
                    className="description-cell"
                  >
                    <strong>
                      Description
                    </strong>
                  </td>
                  <td>
                     
                    bWFpbmx5IGxpYmVydHkgZG9tYWluIGJsb2NrIHNlZWluZyB+NTAlIG9mIGFkZHJlc3NlcyBlbmdhZ2luZyBTRyB3aWRl
                     
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div
            className="detail-info"
          >
            <table
              className="table-wrap"
            >
              <tbody>
                <tr>
                  <td>
                    <strong>
                      Bounce ID
                    </strong>
                  </td>
                  <td>
                     
                    504
                     
                  </td>
                </tr>
                <tr>
                  <td>
                    <strong>
                      Response Code
                    </strong>
                  </td>
                  <td>
                     
                    550
                     
                  </td>
                </tr>
                <tr>
                  <td>
                    <strong>
                      Enhanced Code
                    </strong>
                  </td>
                  <td>
                     
                    N/A
                     
                  </td>
                </tr>
              </tbody>
            </table>
            <table
              className="table-wrap"
            >
              <tbody>
                <tr>
                  <td>
                    <strong>
                      Regex
                    </strong>
                  </td>
                  <td>
                     
                    no MX record for domain
                     
                  </td>
                </tr>
                <tr>
                  <td>
                    <strong>
                      Priority
                    </strong>
                  </td>
                  <td>
                     
                    N/A
                     
                  </td>
                </tr>
                <tr>
                  <td>
                    <strong>
                      Bounce Action
                    </strong>
                  </td>
                  <td>
                     
                    no_action
                     
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="mako-grid__row changelog-container"
  >
    <div
      className="mako-grid__col-10 col-start-2"
    >
      <div
        data-test="changelog"
      >
        <div
          className="mako-grid__row"
          id="changelog-title"
        >
          <div
            className="mako-grid__col-1 col-start-1"
          >
            <h2>
              Changelog
            </h2>
          </div>
          <div
            className="mako-grid__col-2 col-start-11 changelog-csv sg-right"
          >
            <a
              download="changelog.csv"
              href="data:text/csv;charset=utf-8,﻿\\"id\\",\\"response_code\\",\\"enhanced_code\\",\\"regex\\",\\"priority\\",\\"description\\",\\"bounce_action\\",\\"user_id\\"
\\"504\\",\\"550\\",\\"\\",\\"no MX record for domain\\",\\"0\\",\\"bWFpbmx5IGxpYmVydHkgZG9tYWluIGJsb2NrIHNlZWluZyB+NTAlIG9mIGFkZHJlc3NlcyBlbmdhZ2luZyBTRyB3aWRl\\",\\"no_action\\",\\"0\\""
              onClick={[Function]}
              target="_self"
            >
              Export as CSV
            </a>
          </div>
        </div>
        <table
          className="table-wrap changelog-table"
        >
          <thead>
            <tr>
              <th
                className="row-date"
                onClick={[Function]}
              >
                Date
              </th>
              <th
                className="row-user"
                onClick={[Function]}
              >
                User
              </th>
              <th
                className="row-commit"
                onClick={[Function]}
              >
                Commit Message
              </th>
              <th
                className="row-actions actions-align"
                onClick={[Function]}
              >
                Actions
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                <div
                  style={
                    Object {
                      "position": "relative",
                    }
                  }
                >
                  <span
                    className="badge currentBadge"
                  >
                    Current
                  </span>
                </div>
                <td>
                  Invalid date
                </td>
              </td>
              <td>
                N/A
              </td>
              <td>
                N/A
              </td>
              <td>
                 
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div
    className="mako-grid__row"
  >
    <div
      className="mako-grid__col-10 col-start-2"
    >
      <div
        className="pagination pagination-container"
      >
        <a
          className="btn btn-secondary btn-small pagination-prev is-invisible"
          role="button"
          tabIndex="0"
        >
          Prev
        </a>
        <div
          className="pagination-links"
        >
          <a
            className="pagination-link is-active"
            onClick={[MockFunction]}
            onKeyDown={[MockFunction]}
            role="button"
            tabIndex="0"
            value={1}
          >
            1
          </a>
        </div>
        <a
          className="btn btn-secondary btn-small pagination-next is-invisible"
          role="button"
          tabIndex="0"
        >
          Next
        </a>
      </div>
    </div>
  </div>
</div>
`;
